### DISCUSSION TITLE
PyTorch Computer Vision: Video121: Why kernel_size=2 in MaxPool2d?

### DISCUSSION THREAD

**Discussion by karthikbhandary2** [mrdbourke/pytorch-deep-learning#322] [Q&A]
I just wanted to know how we decided on setting the kernel size to 2 for maxpoo2dl. I also want to know how we do that for others as well!

**Comment by MananShukla7** [comment]
I think it depends on the dimensions or size of our image and no of layers our cnn has,Since we have images of size 28x28 
kernel of size 2 is better as it discards half of activations .
After layer 1: image is of dim 14x14
aftet layer 2:Reduced to 7x7
![image](https://user-images.githubusercontent.com/68244767/222895721-9318f14c-8db6-403d-b489-57bac1913531.png)

If we use kernel bigger kernel then, its not gonna be able to detect features efficiently as the dimensions will be too  much reduced 

after doing kernel size-3
![image](https://user-images.githubusercontent.com/68244767/222895900-e077768b-d9b6-4ed9-8056-df15b396cffa.png)

As i said, i think it depends on the dimensions, if we had an image with lets say 64x64 dim and with only 2 layers, then downsampling with bigger size kernel can help in identifying larger or aggregated features of image
